src/posix/platform/netif.cpp

```diff
@@ -384,12 +420,23 @@ static void UpdateUnicastLinux(const otIp6AddressInfo &aAddressInfo, bool aIsAdd
         AddRtAttr(&req.nh, sizeof(req), IFA_CACHEINFO, &cacheinfo, sizeof(cacheinfo));
     }
 
-#if OPENTHREAD_POSIX_CONFIG_NETIF_PREFIX_ROUTE_METRIC > 0
-    if (aAddressInfo.mScope > ot::Ip6::Address::kLinkLocalScope)
+#if OPENTHREAD_POSIX_CONFIG_INSTALL_OMR_ROUTES_ENABLE
+    if (IsOmrAddress(aInstance, aAddressInfo))
```

Comment:
Adding otIp6IsOmrAddress and otNetDataIsValidOmrPrefix looks good to me. 

Maybe we only need `otNetDataIsValidOmrPrefix` since `otIp6IsOmrAddress` can be easily replaced by constructing the IPv6 prefixf from `aAddressInfo` and then calling `otNetDataIsValidOmrPrefix`.

Thoughts?

I think I can add a `otNetDataIsValidOmrPrefix` first.

Reply:
otNetDataIsValidOmrPrefix added.